Sources/SpeziHealthKit/Health Data Collection/HealthDataCollectorDeliverySetting.swift

```diff
@@ -0,0 +1,76 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// Determines the data delivery settings for any ``HealthKitDataSource`` used in the HealthKit module.
+public enum HealthDataCollectorDeliverySetting: Hashable, Sendable {
+    /// The HealthKit data is manually collected when the ``HealthKit/triggerDataSourceCollection()`` function is called.
+    case manual(saveAnchor: Bool = true)
+    /// The HealthKit data is collected based on the ``Start`` and constantly listens to updates while the application is running.
+    /// If `saveAnchor` is enabled the `HKQueryAnchor` is persisted across multiple application launches using the user defaults.
+    case anchorQuery(Start = .automatic, saveAnchor: Bool = true)
```

Comment:
As the manual setting now also includes the option to save the anchor we might want to rename this something like `continuousQuery`, `continuous`, or something similar? As we don't name `background` `backgroundQuery`, it might make more sense to have `manual`, `continuous`, and `background`?

Reply:
Good points, we should ensure that the terminology is aligned across all parts of the package.

I don't think I have better names than `manual`, `continuous`, and `background` at this point but we can continue to think about them as we get closer to a 1.0 release. We might want to merge the PR as it is sooner than later to avoid larger and larger diffs.

Reply:
agreed. my main focus at the moment is to clean up the implementation, restructure the files a bit, write tests and then get it merged.
IMO, the public API and feature set is currently in a good state and there aren't any changes i want to make in that regard (though i might end up removing some functionality from the HealthChart, so that the first version is as stable and usable as possible; it can always get added back later)

Reply:
I haven't renamed these yet, but i think calling it `continuous`, `background`, and `manual` is good. i'll change it to that before the PR is merged, and add deprecated aliases so that old code can migrate 👍